spirv_glsl.cpp

```diff
@@ -3542,25 +3542,43 @@ string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtyp
 			require_extension_internal("GL_ARB_shader_texture_lod");
 	}
 
+	std::string type_prefix = imgtype.image.depth ? "shadow" : "texture";
+
 	if (op == "texture")
-		return join("texture", type);
+		return join(type_prefix, type);
 	else if (op == "textureLod")
 	{
 		if (use_explicit_lod)
-			return join("texture", type, is_legacy_es() ? "LodEXT" : "Lod");
+			return join(type_prefix, type, is_legacy_es() ? "LodEXT" : "Lod");
 		else
-			return join("texture", type);
+			return join(type_prefix, type);
 	}
 	else if (op == "textureProj")
-		return join("texture", type, "Proj");
+		return join(type_prefix, type, "Proj");
 	else if (op == "textureGrad")
-		return join("texture", type, is_legacy_es() ? "GradEXT" : is_legacy_desktop() ? "GradARB" : "Grad");
+		return join(type_prefix, type, is_legacy_es() ? "GradEXT" : is_legacy_desktop() ? "GradARB" : "Grad");
 	else if (op == "textureProjLod")
 	{
 		if (use_explicit_lod)
-			return join("texture", type, is_legacy_es() ? "ProjLodEXT" : "ProjLod");
+			return join(type_prefix, type, is_legacy_es() ? "ProjLodEXT" : "ProjLod");
+		else
+			return join(type_prefix, type);
+	}
+	else if (op == "textureLodOffset")
+	{
+		if (use_explicit_lod)
+			return join(type_prefix, type, "LodOffset");
 		else
-			return join("texture", type);
+			return join(type_prefix, type);
+	}
+	else if (op == "textureProjGrad")
+		return join(type_prefix, type, "ProjGrad");
```

Comment:
Does this exist in legacy GLSL? Wouldn't it need EXT/ARB like "GradARB" at the very least?

Reply:
This exists either as ARB suffixed version through `ARB_shader_texture_lod`, or through `EXT_gpu_shader4` without suffix. I'm not really sure how this should be handled. Is there any precedence on how to handle functionality that is provided by two extensions?

Reply:
I think ARB_shader_texture_lod is already used, so you can reuse that.